### DISCUSSION TITLE
Failing to re-define reference list of iso690 style

### DISCUSSION THREAD

**Discussion by pabloabur** [olsak/OpTeX#166] [Q&A] [answered]
Hi,

I am trying to print all the authors in a LASTNAME, Firstname(s) format (instead of just the first author, as is the standard for iso690). I tried to edit `\_otherauthorformat`, but nothing changed after compilation. In fact, editing `\_firstauthorformat` also does not do anything. Does anybody know what I am missing here?

MWE
test.tex:
```
\fontfam[LMfonts]


\_def\_otherauthorformat{%
    \_upper{\_Lastname}\_bprintc\_Firstname{, *}\_bprintc\_Von{ *}\_bprintc\_Junior{, *}%
}


Citation \cite[ref].

\bigskip
\usebib/s (iso690) bibsample
\bye
```
bibsample.bib
```
@ARTICLE{ref,
    title="title",
    author="Lastname1, Firtname1 and Lastname2, Firstname2",
    year="2024"
    }
```

The output is `Lastname1, Firtname1, and Firstname2 Lastname2. title. 2024.`
What I expected was `Lastname1, Firtname1, and Lastname2, Firstname2. title. 2024.`

**Comment by olsak** [comment, marked as answer]
You have to use `\bibtexhook`, because the macros from the bib-style file (bib-iso690.opm in this example] are read in a group when the `\usebib` macro is processing. So, your definition is re-defined at this moment. But there is `\bibtexhook` token list where a user can put all re-definitions, because `\bibtexhook` is processed after the bib-style file is read. Your example can look like this:

```
 \fontfam[LMfonts]

\bibtexhook{
   \_def\_otherauthorformat{%
       \_upper{\_Lastname}\_bprintc\_Firstname{, *}\_bprintc\_Von{ *}\_bprintc\_Junior{, *}%
   }
}

Citation \cite[ref].

\bigskip
\usebib/s (iso690) bibsample
\bye
```
Maybe, you didn't find the mention about `\bibtexhook` in OpTeX documentation. Please, can you suggest where to put it in order to avoid the confusion of future users?

**Reply by pabloabur** [reply]
Oh I see, thanks.

Now that you mentioned I found `\bibtexhook` in the documentation and the explanation is clear, but I totally missed that. I guess I would have found it if it were in Section **2.32.6 Usage of the bib-iso690 style**. Somewhere around **sorting** or **languages** would make sense to me.

**Reply by olsak** [reply]
I see that the `\bibtexhook` is only documented in the "Parameters" section and it is not enough. My idea is to add the mention about `\bibtexhook` to the chapter 2.32.2 "The `\usebib` command" too.

**Reply by pabloabur** [reply]
Yes, I think that would work.
